botorch/acquisition/proximal.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+A wrapper around AquisitionFunctions to fix certain features for optimization.
+This is useful e.g. for performing contextual optimization.
+"""
+
+from __future__ import annotations
+
+import torch
+from botorch.acquisition.analytic import AnalyticAcquisitionFunction
+from botorch.utils import t_batch_mode_transform
+from torch import Tensor
+from torch.nn import Module
+
+
+class ProximalAcquisitionFunction(AnalyticAcquisitionFunction):
+    """A wrapper around AquisitionFunctions to add proximal weighting of the acquisition function. Acquisition
+    function is weighted via a squared exponential centered at the last training point, with varying lengthscales
+    corresponding to 'proximal_weights'. Can only be used with single batch analytical acquisition functions.
+
+    Example:
+        >>> model = SingleTaskGP(train_X, train_Y)  # d = 2
+        >>> EI = ExpectedImprovement(model, best_f=0.0)
+        >>> proximal_weights = torch.ones(d)
+        >>> EI_proximal = ProximalAcquisitionFunction(EI, proximal_weights)
+        >>> eip = EI_proximal(test_X)  # d' = 3
```

Comment:
Why is `d' = 3` here? I might be missing something but looks like you use the same `d` below.